bittensor/_neuron/text/core_server/__init__.py

```diff
@@ -427,18 +432,25 @@ def synapse_check(self, synapse, hotkey):
                 return False
             
         elif synapse.synapse_type == bittensor.proto.Synapse.SynapseType.TEXT_CAUSAL_LM:
-
-            if self.metagraph.S[incoming_uid] < self.config.neuron.causallm_stake:
+            batch_size, sequence_len  =  inputs_x[0].size()
+            if (self.metagraph.S[incoming_uid] < self.config.neuron.causallm_stake) \
+                and (batch_size > self.config.neuron.max_batch_size) \
+                and (sequence_len > self.config.neuron.max_sequence_len):
                 return False
 
         elif synapse.synapse_type == bittensor.proto.Synapse.SynapseType.TEXT_CAUSAL_LM_NEXT:
-
-            if self.metagraph.S[incoming_uid] < self.config.neuron.causallmnext_stake:
+            batch_size, sequence_len  =  inputs_x[0].size()
+            if (self.metagraph.S[incoming_uid] < self.config.neuron.causallmnext_stake) \
+                and (batch_size > self.config.neuron.max_batch_size) \
+                and (sequence_len > self.config.neuron.max_sequence_len):
                 return False
 
         elif synapse.synapse_type == bittensor.proto.Synapse.SynapseType.TEXT_SEQ_2_SEQ:
-
-            if (self.metagraph.S[incoming_uid] < self.config.neuron.seq2seq_stake) and (self.metagraph.S[incoming_uid,  self.uid]):
+            batch_size, sequence_len  =  inputs_x[0].size()
+            if (self.metagraph.S[incoming_uid] < self.config.neuron.seq2seq_stake) \
+                and (batch_size > self.config.neuron.max_batch_size) \
+                and (sequence_len > self.config.neuron.max_sequence_len) \
+                and (self.metagraph.W[incoming_uid,  self.uid]):
                 return False     
         else:
             return False
```

Comment:
If the synapse isn't in the known synapse types, maybe we should raise an error or warning instead? It could indicate someone's server is out of date, for instance.

Reply:
Fixed as well